### PR TITLE
Add FormBuilderIteratorTest and provide php-ini setting for precision

### PR DESCRIPTION
**Pull request by mikemeier**
* Add phpunit to composer.json
* Test-Template-Files from sonata are generated with a php.ini precision setting of 8, so lets run php also with this predefined setting
* FormBuilder of Symfony can be extended (for example: https://github.com/egeloen/ivory-ordered-form/blob/master/src/Builder/OrderedFormBuilder.php) and this will break sonata admin FormBuilderIterator because the private property "children" is not available

### CODE REVIEW

File: Tests/Util/FormBuilderIteratorTest.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Sonata\AdminBundle\Tests\Controller;
```
Comment (pulzarraider) [inline Tests/Util/FormBuilderIteratorTest.php]: This namespace is wrong.

**Comment by core23** [comment]
Please rebase @mikemeier

**Comment by mikemeier** [comment]
@core23 rebase done

**Comment by dominikzogg** [comment]
@mikemeier finaly ;-)

**Comment by core23** [comment]
Could you please squash your commit's and have a look at travis

File: Util/FormBuilderIterator.php [hunk 2]
```diff
@@ -65,12 +65,7 @@ public function __construct(FormBuilderInterface $formBuilder, $prefix = false)
      */
     private static function getKeys(FormBuilderInterface $formBuilder)
     {
-        if (!self::$reflection) {
```
Comment (core23) [inline Util/FormBuilderIterator.php]: Why did you remove this?

File: composer.json [hunk 3]
```diff
@@ -44,7 +44,9 @@
         "symfony/yaml": "~2.3|~3.0",
         "symfony/security-acl": "~2.4|~3.0",
         "sonata-project/intl-bundle": "dev-master",
-        "symfony/phpunit-bridge": "~2.7|~3.0"
+        "symfony/phpunit-bridge": "~2.7|~3.0",
```
Comment (core23) [inline composer.json]: They are not required for this PR.

File: phpunit.xml.dist [hunk 4]
```diff
@@ -30,4 +30,8 @@
         </whitelist>
     </filter>
 
+    <php>
+        <ini name="precision" value="8"/>
+    </php>
+
```
Comment (core23) [inline phpunit.xml.dist]: Why?

File: Util/FormBuilderIterator.php (hunk 2 above)
Comment (mikemeier) [inline Util/FormBuilderIterator.php]: I removed the whole reflection part, because sonata access a private property "children". Not every FormBuilderIterator has this property. Its described in my first comment for this PR.

File: composer.json (hunk 3 above)
Comment (mikemeier) [inline composer.json]: That was because of a merge after rebase, should I delete it? With my commit, it would be already resolved...

File: phpunit.xml.dist (hunk 4 above)
Comment (mikemeier) [inline phpunit.xml.dist]: Described in my description text for this PR. If my php.ini uses a precision of 12, the already written unit tests will fail otherwise.

File: composer.json (hunk 3 above)
Comment (core23) [inline composer.json]: Yes, please remove this because phpunit isn't explicitly required for this.

File: phpunit.xml.dist (hunk 4 above)
Comment (core23) [inline phpunit.xml.dist]: Can you please show me where the sonata admin uses the precison value? Haven't found any location where we rely on this value.

Comment (mikemeier) [inline phpunit.xml.dist]: Just let your tests run with precision of 5:

```
1) Sonata\AdminBundle\Tests\Twig\Extension\SonataAdminExtensionTest::testRenderListElement with data set #21 ('<td class="sonata-ba-list-fie... </td>', 'number', 10.746135000000001, array())
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-<td class="sonata-ba-list-field sonata-ba-list-field-number" objectId="12345"> 10.746135 </td>
+<td class="sonata-ba-list-field sonata-ba-list-field-number" objectId="12345"> 10.746 </td>
```

Comment (OskarStark) [inline phpunit.xml.dist]: hmm IMO we have to rewrite the other test (i mean the failing example), what do you think @core23

Comment (mikemeier) [inline phpunit.xml.dist]: There are a lot more of them @OskarStark (8 to be precise). And you cant rewrite them correctly. If your php.ini uses precision 200, you will get other results than with precision 5. So the correct way in my opinion is to write your tests with the knowledge of the php.ini precision (thats what I did with the xml change). If you're not comfortable with precision "8" we could also use "100" and then round the values before comparing in the unit test.

Comment (OskarStark) [inline phpunit.xml.dist]: sounds reasonable to me @mikemeier

File: Tests/Util/FormBuilderIteratorTest.php [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        $this->builder->add('name', 'text');
+        $iterator = new FormBuilderIterator($this->builder);
+        $this->assertTrue($iterator->hasChildren());
+    }
+}
+
+class TestFormBuilder extends FormBuilder
+{
+
+}
```
Comment (OskarStark) [inline Tests/Util/FormBuilderIteratorTest.php]: please execute `make cs`

**Comment by OskarStark** [comment]
@mikemeier can you please check why travis is failing?

**Comment by core23** [comment]
Please rebase @mikemeier
